x-pack/plugins/security_solution/public/common/hooks/eql/api.ts

```diff
@@ -29,3 +39,59 @@ export const validateEql = async ({
     signal,
   });
 };
+
+interface AggsParams extends EqlValidationRequest {
+  data: DataPublicPluginStart;
+  interval: Unit;
+  fromTime: string;
+  toTime: string;
+  signal: AbortSignal;
+}
+
+export const getEqlPreview = async ({
+  data,
+  index,
+  interval,
+  query,
+  fromTime,
+  toTime,
+  signal,
+}: AggsParams): Promise<EqlPreviewResponse> => {
+  try {
+    const response = await data.search
+      .search<EqlSearchStrategyRequest, EqlSearchStrategyResponse<EqlSearchResponse<Source>>>(
+        {
+          params: {
+            // @ts-expect-error allow_no_indices is missing on EqlSearch
+            allow_no_indices: true,
+            index: index.join(),
+            body: {
+              filter: {
+                range: {
+                  '@timestamp': {
+                    gte: toTime,
+                    lte: fromTime,
+                  },
+                },
+              },
+              query,
+              size: 50,
+            },
+          },
+        },
+        {
+          strategy: 'eql',
+          abortSignal: signal,
+        }
+      )
+      .toPromise();
+
+    if (query.indexOf('sequence') !== -1) {
```

Comment:
We could take advantage of this helper here to determine if the query is a sequence query https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/common/detection_engine/utils.ts#L20

Reply:
Thanks! Fixed - https://github.com/elastic/kibana/pull/78985/commits/e7462c8a45d4cbe00247452d581f392055d30134